QuestionBot/QuestionBot/Model/InMemoryStore.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace QuestionBot.Model {
+
+    public class InMemoryStore : IStore {
+        private IList<IRecord> allRecords = new List<IRecord>();
+        private static int _questionId = 0;
+
+        public IRecord CreateRecord(string question){
+            _questionId++;
+
+            Record newQuestion = new Record(_questionId, question, DateTime.Now);
+            allRecords.Add(newQuestion);
+
+            return newQuestion;
+        }
+
+        public bool TryUpdateRecord(int id, string answer){
```

Comment:
If you refer to the notes from the error handling session, we're *almost* following the pattern, just missing one last thing. https://gist.github.com/AlexBedley/a1f17121b98d356a512c6d4192d6e43d#file-errorhandling-cs-L63